### PR TITLE
[Email] Optimize email flow with read more

### PR DESCRIPTION
**Pull request by DingmaomaoBJTU**
## Description
Optimize email flow
1. Enable 'ReadMore' flow to current ShowEmail/SendEmail/ForwardEmail flow.
2. Enable 'ReadMore' flow to current recipient select step.
3. Fix some bugs related to next/previous page for email list and contact

## Related Issue
Add read more support for email and recipient list #339

## Testing Steps
For email list read more:
1. User: what is my emails?
2. Bot: show list of [6] emails, read out first [3] one
3. User: read more
4: Bot: show the rest of [3] emails, and read them out
5. User: read more
6: Bot:  show next page - list of [6] emails, read out first [3] one

For recipient list read more:
1. User: send an email to qiong
2. Bot: show list of [6] contacts, read out first [3] one
3. User: read more
4: Bot: show the rest of [3] contacts, and read them out
5. User: read more
6: Bot:  show next page - list of [6] contacts, read out first [3] one

The display number [6] and read number [3] is configurable.

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json [hunk 1]
```diff
@@ -48,6 +48,8 @@
         "IPA.Timezone"
       ],
       "configuration": {
+        "maxReadSize": "3",
+        "maxDisplaySize": "6"
```
Comment (feich-ms) [inline solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: Where are there two parameters used?

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Util/CommonUtil.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.Bot.Solutions.Resources;
+
+namespace Microsoft.Bot.Solutions.Util
+{
+    public class CommonUtil
+    {
+        public static readonly double ScoreThreshold = 0.5f;
+
+        public static readonly int MaxReadSize = 3;
```
Comment (feich-ms) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Util/CommonUtil.cs]: I think it is not needed to limit the size here. It should be up to users.

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Util/CommonUtil.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+        public static readonly int MaxDisplaySize = 6;
+
+        public static bool IsReadMoreIntent(string userInput)
+        {
+            return userInput.ToLowerInvariant().Contains(CommonStrings.More);
+        }
```
Comment (feich-ms) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Util/CommonUtil.cs]: I think we should move this to the function of parsing lu entities.

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json (hunk 1 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: I think we should remove 'max' from the parameter names here and just use ReadSize and DisplaySize.

**Review by feich-ms — APPROVED** [review]

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json (hunk 1 above)
Comment (DingmaomaoBJTU) [reply solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: Good suggestion! Fixed, thanks!~

**Review by darrenj — CHANGES_REQUESTED** [review]
- Should ConfigData.cs not be within the Email skill rather than bot.solutions, not sure it's shared?
- As per other PR feedback from Lauren, looks like the changes to CommonStrings are only used by the email skill and not shared, we should place this into the Email commonstrings resource file?

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Data/ConfigData.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Bot.Solutions;
+
+namespace Microsoft.Bot.Solutions.Data
+{
+    public class ConfigData
+    {
+        private static int _maxReadSize = Util.CommonUtil.MaxReadSize;
+        private static int _maxDisplaySize = Util.CommonUtil.MaxDisplaySize;
+
+        public static int MaxReadSize
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Data/ConfigData.cs]: no good comes out of static properties. You can either use the settings from skillconfig directly in wherever needed, or make this a singleton so you can easily mock it when testing.

**Review by lzc850612 — CHANGES_REQUESTED** [review]

**Comment by DingmaomaoBJTU** [comment]
@darrenj These strings I just added are shared ones, and calendar skill is waiting to use them. So I will still keep them in common~ And for ConfigData.cs, it can be shared, and I will make it singleton to ensure better experience~

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Data/ConfigData.cs (hunk 4 above)
Comment (DingmaomaoBJTU) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Data/ConfigData.cs]: Make this a singleton~ Thanks!

**Review by lzc850612 — APPROVED** [review]

**Review by darrenj — APPROVED** [review]

**Comment by darrenj** [comment]
@DingmaomaoBJTU Looks like there is a build failure, can you resolve before merging please?

**Comment by DingmaomaoBJTU** [comment]
@darrenj Sure, fixed! Sorry for that~~

**Pull request merged**